### PR TITLE
Feature: Simulate Experiences and TTS output with IVR supporting DTMF 

### PR DESCRIPTION
**Pull request by nirvanatikku**
Simulate experiences: Visual VA and IVR with TTS output and DTMF input (use-case dep)

* DLG: Integrate TTS output support with play queue
* DLG: Support NO_INPUT event triggers (for IVR)
* DLG: Augment execute handler to support DTMF & Events
* LOG: Events rendering (nlg vs visual) based on experience
* Chat: Keypad input 
* Chat: Support `width` property on `img` tag when Rich Text
* Shared: Improve scope handling
* Profile: Revamped with note around client secret handling & support session storage

### CODE REVIEW

File: src/components/shared.js [hunk 1]
```diff
@@ -86,6 +86,41 @@ export const LANG_EMOJIS = {
...
+const EXPERIENCE_TYPES = {
+  ivrTextWithTts: {
+    playTTS: true,
+    isOutputHTML: false,
+    isOutputSSML: true,
+    bindTimeouts: true,
+    dtmfInput: true
+  },
+  ivrTextWithSSML: {
+    playTTS: false,
```
Comment (jbf154) [inline src/components/shared.js:98]: Any reason you wouldn't want to play TTS with SSML?

File: src/components/dlgaas.js [hunk 2]
```diff
@@ -622,34 +810,98 @@ export default class DLGaaS extends BaseClass {
...
+            </div>
+
             <div className={(this.isStandalone() ? `col-12` : `col-4 offset-md-1`) + ` bg-light rounded-3 px-4 py-4`}>
+
               <form className="form" onSubmit={(evt) => {this.go(); evt.preventDefault();}}>
                 <h4 className="w-100">Configuration</h4>
                 <p>
                   See <a target="_blank" rel="noreferrer" href="https://docs.mix.nuance.com/languages/?src=demo#languages-and-voices">Languages and Voices</a>
                 </p>
+                {SIMULATED_EXPERIENCES(this.state.simulateExperience).playTTS ? (
```
Comment (jbf154) [inline src/components/dlgaas.js:882]: This <select/> is always empty for me.

File: src/components/dlgaas.js [hunk 3]
```diff
@@ -471,33 +563,126 @@ export default class DLGaaS extends BaseClass {
...
+      clearTimeout(this.recoTimeout)
+    }
+    this.recoTimeout = setTimeout(() => {
+      Promise.resolve().then(() => {
+        this.execute(null, false, {
+            type: 'NO_INPUT',
+            message: `Standard reco timeout kicked in after ${mainTimeout}.`,
+        })
+      })
+    }, parseInt(mainTimeout.substring(0, mainTimeout.length-2))) // chop the `ms`
```
Comment (jbf154) [inline src/components/dlgaas.js:621]: Timeout doesn't always end in `ms`.  Update to use regex.

File: src/components/shared.js [hunk 4]
```diff
@@ -86,6 +86,41 @@ export const LANG_EMOJIS = {
...
 
+const EXPERIENCE_TYPES = {
+  ivrTextWithTts: {
+    playTTS: true,
+    isOutputHTML: false,
+    isOutputSSML: true,
+    bindTimeouts: true,
+    dtmfInput: true
+  },
+  ivrTextWithSSML: {
```
[the remark this answers is not in the document]
Comment (nirvanatikku) [reply src/components/shared.js:97]: Keeping this as-is for now and will be revised in a future update.

File: src/components/dlgaas.js (hunk 2 above)
Comment (nirvanatikku) [reply src/components/dlgaas.js:882]: To be fixed in future release (pulling things in to true-up with the pending PRs).

File: src/components/dlgaas.js (hunk 3 above)
Comment (nirvanatikku) [reply src/components/dlgaas.js:621]: Noted - to be fixed in future release.

**Review by sunil-thorat — APPROVED** [review]

File: src/components/shared.js (hunk 1 above)
Comment (nirvanatikku) [reply src/components/shared.js:98]: Re-approaching in future release.

**Pull request merged**